tests/installation/validation/iscsi_ibft.pm

```diff
@@ -0,0 +1,171 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test module to verify iBFT configuration
+# iSCSI Boot Firmware Table (iBFT) is a mechanism for the iSCSI tools to
+# extract from the machine NICs the iSCSI connection information so that they
+# can automagically mount the iSCSI share/target.  Currently the iSCSI
+# information is hard-coded in the initrd. The /sysfs entries are read-only
+# one-name-and-value fields.
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use Scalar::Util 'looks_like_number';
+use Data::Dumper;
+
+my $ibft_expected = {
+    ethernet => {
+        flags       => 3,
+        prefix_len  => 24,
+        subnet_mask => '255.255.255.0',
+        gateway     => '10.0.2.2',
+        ip_addr     => '10.0.2.15',
+        mac         => '',
+    },
+    initiator => {
+        initiator_name => 'iqn.2010-04.org.ipxe:00000000-0000-0000-0000-000000000000',
+        flags          => 3,
+    },
+    target => {
+        target_name => '',
+        flags       => 3,
+        port        => '3260',
+        ip_addr     => '',
+        chap_type   => 0
+    },
+    acpi_header => {
+        oem_table_id => 'iPXE',
+        oem_id       => 'FENSYS',
+        signature    => 'iBFT'
+    },
+    backstore => {
+        model  => q/'FILEIO|IBLOCK'/,
+        vendor => q/'LIO-ORG'/
+    }
+};
+
+$ibft_expected->{target}->{target_name} = get_required_var('NBF');
```

Comment:
You can actually inject those in the initialization.
I've tried with
```perl
sub lll {
    return 9;
}

my $ibft_expected = {
    ethernet => lll(),
};

print $ibft_expected->{ethernet};
```
and it works fine

Reply:
That is completely true however if I used it as a bash argument:
http://eris.suse.cz/tests/12439#step/iscsi_ibft/17

Reply:
ah wait.... I used construct 
```perl
my $hash = { something => sub { return get_required_var('NBF'); } }
```